### DISCUSSION TITLE
v0.9 Beta 1

### DISCUSSION THREAD

**Discussion by HaSa1002** [Libre-TrainSim/Libre-TrainSim#339] [General]
A curated changelog will follow for the release of 0.9. In the meantime, GitHub provides us with an auto-generated changelog. As the tracks still need to be updated this release only includes tutorials.

The biggest new features are an in-game editor, new signals, backwards driving, high-scores, new tutorials, a rail map, new languages, improved mod support, controller support, improved UI, and more. There have been a ton of bug fixes, too.

<details>
<summary>Click to read the changelog</summary>

## What's Changed
* Fixed small grammar mistake in Ingame Translations file by @akuspel in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/88
* Master to WIP Ingame Editor by @Jean28518 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/89
* WIP: Translation by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/91
* master to ingame-editor by @Jean28518 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/97
* jSaveModule: fix: make_dir_recursive by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/101
* Implement Free Look and Zoom in Cabin View by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/92
* Fix error messages produced by the engine by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/102
* Remove duplicated stoneplate import file by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/100
* Rail Map by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/99
* Add Controller Input by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/110
* Allow loading of addons from user://addons by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/114
* Add Ingame track editor by @Jean28518 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/115
* Fix: Configuration broken by Cache Implementation by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/121
* Restore map functionality by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/120
* bake_route fixes by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/119
* Refactor Signals by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/122
* Refactor platformSide by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/127
* Add fullscreen shortcut by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/129
* About page by @bodamat in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/128
* fix: Signal VisualInstance wrong initial state by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/136
* Driving Backwards by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/95
* Fix jEssentials script error in 3.3.3 by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/139
* Fix jTools delayed calls are causing null references by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/141
* Add more Signals by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/125
* Fix signals and speed limits trying to set themselves to rails, without being in tree by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/140
* Enforce LF in repo and tell GH about GDScript by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/160
* Fix memory leaks by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/159
* Add CI by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/158
* fix normals, enable backface culling by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/156
* Fix save paths by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/155
* fix SpeedLimit and WarnSpeedLimit scene paths by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/149
* Fix CI run on linux by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/161
* Fix Typo by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/152
* Fix trains embedded meshes and wrong Wheel direction by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/153
* Overhaul JFR1 Train by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/162
* Remove trailing whitespaces and tabs by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/144
* Fix ProjectSettings.save() being called by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/147
* Refactor menus by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/133
* Improve editor camera by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/172
* Better Grass by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/175
* Add better logging by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/145
* Add: PZB Safety System by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/163
* Add threaded loading to loading screen by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/154
* add drag and drop for rails in editor :) by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/164
* Update links by @Jean28518 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/198
* Fix gras meshes not getting deleted by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/195
* Fix `!owner.valid` errors caused by chunk system and `load_world` by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/186
* Fix `handle_drag_mode` connecting multiple times to same signals by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/184
* Prevent scripts from running in editor by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/196
* Fix select collider not working with signals by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/189
* Fix editor not being able to swtich to main menu by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/188
* Fix camera in editor by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/177
* Fix stolen mouse input in editor by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/183
* performance hotfix by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/204
* Fix selection color of buttons and ItemLists by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/205
* Fix tutorials not getting properly exported by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/206
* Refactor Project Structure for Modding by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/210
* fix world loading by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/214
* fix pzb screen desync by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/215
* hide TrackObject tab when a non-rail is selected by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/216
* Add Highlight postprocessing to Editor selection (remove bouncing animation) by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/208
* Fix sounds engine sounds playing when game is paused by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/207
* automatically add poles when rail is electric by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/217
* Fix: Name Me! folder being created by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/225
* Nicer modding tools by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/226
* fix default rail in new world has no poles by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/234
* fix everything using the wrong audio bus by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/235
* Add mod exporting by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/227
* fix rail logic non existent signal by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/232
* fix rail type leaks by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/231
* Convert editor maps to mods by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/228
* Move camera faster the longer you move by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/237
* fix jsettings close on esc by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/238
* Fix Ingame Editor Export by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/240
* Fix RailLogic setting visiblity in _input by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/243
* fix scenario to signals by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/247
* add editor translations by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/248
* Add difficulty settings by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/250
* fix getPathFromTo by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/246
* fix JFR1 Normals by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/256
* Add spanish translation. by @mdtrooper in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/257
* Update project to 3.4 by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/259
* Add very simple scoring system by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/249
* Fix screenshots not being loaded in builds by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/264
* Fix Shadow Acne by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/263
* Refactor Chunk System by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/262
* Fix a few bugs by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/266
* new imports files by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/269
* fix crawl directory by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/271
* fix chunks not unloading correctly by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/276
* Fix message label readability using blur by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/274
* Enable MSAA x4 for JFR1 screen textures by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/275
* Add Scenario Editor by @Jean28518 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/278
* Improe cabin inside camera script by @Jean28518 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/295
* Redesign Cabin Model by @Jean28518 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/294
* Add bulgarian translation by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/296
* Update links to new url by @Jean28518 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/298
* Fix missing signals and null exception in scenario editor by @Jean28518 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/299
* Disable right mouse button click in free camera ingame by @Jean28518 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/309
* Fix visibilty update of npc trains in unloaded chunks by @Jean28518 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/308
* Fix state after scenario editor by @Jean28518 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/304
* Add: Selection of buildings in the editor by @Jean28518 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/305
* Fix export and test buttons by @Jean28518 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/302
* Add deletion of content and tracks by @Jean28518 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/303
* Fix chunk manager at loading far away from world origin by @Jean28518 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/316
* Fix Hl signal stack overflow by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/321
* Fix translations not getting applied by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/320
* Update CI to 3.4.2 by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/322
* Updated the Bulgarian and English translations by @RacerBG in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/323
* Fix message not getting shown when pressing back in pause menu by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/318
* Fix editor camera not grabbing keyboard focus by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/317
* Remove content import feature by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/319
* Fix german translations by @Jean28518 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/311
* Small editor improvements by @Jean28518 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/324
* Add french translation by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/333
* Fix them passenger system by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/332
* Fix Tutorials by @DasCapschen in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/335
* Fix ingame track export by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/336
* Add new logo by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/267
* Update user data directory by @HaSa1002 in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/338

## New Contributors
* @akuspel made their first contribution in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/88
* @DasCapschen made their first contribution in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/91
* @HaSa1002 made their first contribution in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/100
* @bodamat made their first contribution in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/128
* @mdtrooper made their first contribution in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/257
* @RacerBG made their first contribution in https://github.com/Libre-TrainSim/Libre-TrainSim/pull/323

**Full Changelog**: https://github.com/Libre-TrainSim/Libre-TrainSim/compare/v0.8...v0.9.beta1

</details>

<hr /><em>This discussion was created from the release <a href='https://github.com/Libre-TrainSim/Libre-TrainSim/releases/tag/v0.9.beta1'>v0.9 Beta 1 </a>.</em>

**Comment by CyberFox001** [comment]
Thank everyone for the great work. :heart:

**Comment by CleverLemming1337** [comment]
Hello! This game is really fun, I have already played all the tutorials.
Are the additional tracks like Hainfurt already playable in v0.9 and if so, how can they be added to the game?

Thanks in advance
